docs: new terms for glossary (#613)
o1-labs/docs2: docs/glossary.mdx

### zk-SNARK

> A zero knowledge proof. zk-SNARK is the acronym for zero knowledge succinct non-interactive argument of knowledge. Specific properties of interest in Mina's implementation of SNARKs are succinctness and non-interactivity which allow for any node to quickly verify the state of the network.

Reviewer: Just curious - why did you remove "a type of" from this definition?

> A recursive zero knowledge proof. zk-SNARK is the acronym for zero knowledge succinct non-interactive argument of knowledge. Specific properties of interest in Mina's implementation of SNARKs are succinctness and non-interactivity that allow for any node to quickly verify the state of the network.

Reviewer 2: There are some kinds of recursive zero knowledge proofs that are generally not referred to as SNARKs (the SNARK name is a bit confusing and isn't especially well-defined).

Author: good call, I removed "recursive" 

I got the recursive bit from https://minaprotocol.com/blog/what-are-zk-snarks 

We have a lot of clarity to unmuddy

> A recursive zero knowledge proof. zk-SNARK is the acronym for zero knowledge succinct non-interactive argument of knowledge. Specific properties of interest in Mina's implementation of SNARKs are succinctness and non-interactivity that allow for any node to quickly verify the state of the network.

Author: ```suggestion
A zero knowledge proof. zk-SNARK is the acronym for zero knowledge succinct non-interactive argument of knowledge. Specific properties of interest in Mina's implementation of SNARKs are succinctness and non-interactivity which allow for any node to quickly verify the state of the network.
```
